src/python/tests/test_configuration_manager.py

```diff
@@ -0,0 +1,710 @@
+        except Exception:
+            raise
+
+        await set_category_item_value_entry(category_name='boolean',
+                                            item_name='info', new_value_entry='True')
+        try:
+            async with aiopg.sa.create_engine(_CONNECTION_STRING) as engine:
+                async with engine.acquire() as conn:
+                    async for result in conn.execute(stmt):
+                        assert result[0]['info']['value'] == 'True'
+                        assert result[0]['info']['default'] == 'False'
+        except Exception:
+            raise
+        await self.teardown_method()
+
+    async def test_set_category_item_value_error(self):
+        """
+        Test updating of configuration.value when configuration is not set
+        :assert:
+            As of now the expected behavior is nothing gets returned; however,
+            there is an open issue expecting an error be returned when an update
+            value for nonexistent category is called.
+        """
+        stmt = sa.select([_configuration_tbl.c.value]).select_from(_configuration_tbl).where(
+            _configuration_tbl.c.key == 'boolean')
+        await self.setup_method()
+        await set_category_item_value_entry(category_name='boolean',
+                                            item_name='info', new_value_entry='True')
+        try:
+            async with aiopg.sa.create_engine(_CONNECTION_STRING) as engine:
+                async with engine.acquire() as conn:
+                    async for result in conn.execute(stmt):
+                        assert result is None
+        except Exception:
+            raise
+        await self.teardown_method()
+
+    async def test_get_category_item_value_entry(self):
+        """
+        Test that get_category_item_value_entry works properly
+        :Assert:
+            1. category_value.value gets returned and matches default
+            2. When updating value, the data retrieved for value gets updated
+        """
+        await self.setup_method()
+        await create_category(category_name='boolean', category_description='boolean type',
+                              category_value={'info': {
+                                  'description': 'boolean type with default False',
+                                  'type': 'boolean',
+                                  'default': 'False'}})
+        result = await get_category_item_value_entry(category_name='boolean', item_name='info')
+        assert result == 'False'
+
+        await set_category_item_value_entry(category_name='boolean',
+                                            item_name='info', new_value_entry='True')
+        result = await get_category_item_value_entry(category_name='boolean', item_name='info')
+        assert result == 'True'
+        await self.teardown_method()
+
+    async def test_get_category_item_value_entry_empty(self):
```

Comment:
I really doubt this should be here. we should not check non-existent things there can be 100x many things for each particle!

Reply:
you are absolutely right; however, we should have some level of negative testing. As such, these can either be as part of the existing class (what we currently have), or as an independent class.